pkg/ipam/crd.go

```diff
@@ -337,12 +337,20 @@ func (n *nodeStore) updateLocalNodeResource(node *ciliumv2.CiliumNode) {
 	releaseUpstreamSyncNeeded := false
 	// ACK or NACK IPs marked for release by the operator
 	for ip, status := range n.ownNode.Status.IPAM.ReleaseIPs {
-		if status != ipamOption.IPAMMarkForRelease || n.ownNode.Spec.IPAM.Pool == nil {
```

Comment:
nit on commit msg title: We could probably just shorten it to 

```
Fix CN update order by adding new `released` IPAM state
```

Reply:
Updated the commit message.